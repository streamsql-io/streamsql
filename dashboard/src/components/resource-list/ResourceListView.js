import React from "react";
import { makeStyles } from "@material-ui/core/styles";
import Box from "@material-ui/core/Box";
import MaterialTable from "material-table";

const useStyles = makeStyles((theme) => ({
  root: {},
  table: {
    margin: theme.spacing(4),
  },
}));

export const ResourceListView = ({ title }) => {
  const classes = useStyles();
  return (
    // MaterialTable doesn't support className, so we wrap it in a Box:
    // https://github.com/mbrn/material-table/issues/766
    <Box className={classes.table}>
      <MaterialTable
        title={title}
        columns={[{ title: "Name", field: "name" }]}
        data={[{ name: "test1" }, { name: "test2" }]}
        options={{
          search: true,
        }}
      />
    </Box>
  );
};

export default ResourceListView;
