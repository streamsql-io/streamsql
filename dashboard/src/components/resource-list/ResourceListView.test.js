import React from "react";
// Necessary to get MaterialTable to work correctly.
import "jest-canvas-mock";
import { configure, shallow, mount } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import produce from "immer";

import { ResourceListView, VersionSelector } from "./ResourceListView.js";

configure({ adapter: new Adapter() });

describe("ResourceListView", () => {
  it("passes through title", () => {
    const list = shallow(<ResourceListView title="test" title={"test"} />);
    expect(list.children().props().title).toBe("test");
  });

  it("sets resources to [] by default", () => {
    const list = shallow(<ResourceListView title="test" />);
    expect(list.children().props().data).toEqual([]);
  });

  it("passes through resources", () => {
    const list = shallow(
      <ResourceListView title="test" resources={[{ name: "abc" }]} />
    );
    expect(list.children().props().data).toEqual([{ name: "abc" }]);
  });

  it("makes resources mutable", () => {
    const immutData = produce([], (draft) => {
      draft.push(["abc"]);
    });
    const list = shallow(
      <ResourceListView title="test" resources={immutData} />
    );
    expect(Object.isFrozen(immutData)).toBe(true);
    expect(Object.isFrozen(list.children().props().data)).toBe(false);
  });

  it("sets isLoading when resources isn't set", () => {
    const list = shallow(<ResourceListView title="test" />);
    expect(list.children().props().isLoading).toEqual(true);
  });

  it("sets isLoading when loading", () => {
    const list = shallow(<ResourceListView title="test" loading={true} />);
    expect(list.children().props().isLoading).toEqual(true);
  });

  it("sets isLoading when failed", () => {
    const list = shallow(
      <ResourceListView title="test" loading={false} failed={true} />
    );
    expect(list.children().props().isLoading).toEqual(true);
  });

  describe("VersionSelector", () => {
    it("sets default active version", () => {
      const sel = shallow(<VersionSelector name="abc" versions={["1", "2"]} />);
      expect(sel.children().props().value).toBe("1");
    });

    it("onChange calls setVersion", () => {
      const onChange = jest.fn();
      const name = "abc";
      const clickVal = "2";
      const sel = shallow(
        <VersionSelector
          name={name}
          versions={["1", "2"]}
          setVersion={onChange}
        />
      );
      sel.children().prop("onChange")({ target: { value: clickVal } });
      expect(onChange).toHaveBeenCalledWith(name, clickVal);
    });
  });
});
