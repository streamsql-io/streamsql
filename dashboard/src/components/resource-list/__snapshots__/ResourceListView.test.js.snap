// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResourceListView TagList renders correctly with no active tags 1`] = `
<WithStyles(ForwardRef(Grid))
  container={true}
  direction="row"
>
  <WithStyles(ForwardRef(Chip))
    className="class-here"
    color="default"
    key="a"
    label="a"
    onClick={[Function]}
    variant="outlined"
  />
  <WithStyles(ForwardRef(Chip))
    className="class-here"
    color="default"
    key="b"
    label="b"
    onClick={[Function]}
    variant="outlined"
  />
  <WithStyles(ForwardRef(Chip))
    className="class-here"
    color="default"
    key="c"
    label="c"
    onClick={[Function]}
    variant="outlined"
  />
</WithStyles(ForwardRef(Grid))>
`;
